### PR TITLE
Implemented Search Feature in Filter Menu

### LINKED ISSUE #4367: Implement Search Feature in Filter Menu
# Current Behavior

CleanShot 2024-03-07 at 18 19 14

# Desired Behavior

The filter menu currently lacks a 'Search' function. To optimize keyboard navigation and handle numerous fields effectively, we propose the addition of a search feature at the top

CleanShot 2024-03-07 at 18 23 48

# Figma

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=2979-194754&mode=design&t=LORfESmPs8dJ3RY2-11

### PR DESCRIPTION
**Pull request by rathodvinod5**
Closes #4367

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownFilterSelect.tsx [hunk 1]
```diff
@@ -25,31 +55,50 @@ export const ObjectFilterDropdownFilterSelect = () => {
 
   const setHotkeyScope = useSetHotkeyScope();
 
+  const debouncedSetSearchFilter = useDebouncedCallback(setSearchText, 100, {
```
Comment (Weiko) [inline packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownFilterSelect.tsx:58]: I don't think we need a debounce for this as this is not querying/computing anything, let's not slow down the experience when it's not needed 👍

**Review by Weiko — COMMENTED** [review]
Left a comment @rathodvinod5 otherwise lgtm 👍

**Comment by Weiko** [comment]
LGTM otherwise, I will update this part in another PR 👍 
Thank you for your contribution @rathodvinod5

**Review by Weiko — APPROVED** [review]

**Comment by rathodvinod5** [comment]
@Weiko thanks for the update.

**Comment by rathodvinod5** [comment]
I will try to remove the debounce effect in next PR.

**Pull request merged**